pallets/ds-maps/src/lib.rs

```diff
@@ -0,0 +1,192 @@
+#[derive(Encode, Decode, Clone, Default, Debug, PartialEq, Eq)]
+pub struct Point3D<Coord> {
+    x: Coord,
+    y: Coord,
+    z: Coord,
+}
+impl<Coord> Point3D<Coord> {
+    pub fn new(x: Coord, y: Coord, z: Coord) -> Self {
+        Point3D{x, y, z}
+    }
+}
+
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, Default, Debug, PartialEq, Eq)]
+pub struct Box3D<Point> {
+    point_1: Point,
+    point_2: Point,
+}
+
+impl<Point> Box3D<Point> {
+    pub fn new(point_1: Point, point_2: Point) -> Self {
+        Box3D{point_1, point_2}
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, Default)]
+pub struct Zone<Point> {
+    pub bounding_box: Box3D<Point>,
+    pub zone_type: ZoneType,
+    pub zone_id: u32,
+}
+
+impl<Point> Zone<Point> {
+    pub fn zone_is(&self, zone: ZoneType) -> bool {
+        self.zone_type == zone
+    }
+
+    pub fn new( zone_id: u32, 
+                zone_type: ZoneType, 
+                bounding_box: Box3D<Point> ) -> Self {
+            Zone {
+                bounding_box,
+                zone_type,
+                zone_id,
+            }
+    }
+}
+
+/// Configure the pallet by specifying the parameters and types on which it depends.
+pub trait Trait: accounts::Trait {
+    /// Because this pallet emits events, it depends on the runtime's definition of an event.
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+    // Describe pallet constants.
+    // Lean more https://substrate.dev/docs/en/knowledgebase/runtime/metadata
+    type WeightInfo: WeightInfo;
+    // new types, consider description
+    /// representing a point in space
+    type Point: Default + Parameter;
+    /// guess use u32 for representing global coords, u16 for local
+    type Coord: Default + Parameter;
+}    
+pub trait WeightInfo {
+    fn zone_add() -> Weight;
+}
+
+decl_storage!{
+    // A unique name is used to ensure that the pallet's storage items are isolated.
+    // This name may be updated, but each pallet in the runtime must use a unique name.
+    // ---------------------------------vvvvvvvvvvvv
+    trait Store for Module<T: Trait> as DSMapsModule {
+        // MAX is 4_294_967_295. Change if required more.
+        TotalBoxes get(fn total_boxes): u32;    
+
+        CityMap get(fn map_data): 
+            map hasher(blake2_128_concat) u32 => ZoneOf<T>;
+    }
+}
+pub type ZoneOf<T> = Zone<<T as Trait>::Point>;
+
+// Pallets use events to inform users when important changes are made.
+// https://substrate.dev/docs/en/knowledgebase/runtime/events
+decl_event!(
+    pub enum Event<T>
+    where
+        AccountId = <T as frame_system::Trait>::AccountId,
+        Coord = <T as Trait>::Coord,
+    {
+        // Event documentation should end with an array that provides descriptive names for event parameters.
+        /// TODO add more meta/remove
+        MapInitialized(Coord),
```

Comment:
What does this event mean? What is the meaning of the parameter?

Reply:
Guess, its really better to remove it for now. 
I had some ideas about it, but now its quite useless.